File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;
        int res = INT_MIN;
        while(left < right) {
            res = max(res, min(height[left], height[right]) * (right - left));
            if(height[right] < height[left]) right--;
            else left++;
        }
        return res;
    }
};

File: 11-container-with-most-water/11-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        
        
        //Optimized
        int max = Integer.MIN_VALUE;
        int low = 0;
        int high = height.length - 1;
        while(low <= high) {
            max = Math.max(max, (high - low)*(Math.min(height[low], height[high])));
            if(height[low] < height[high]) {
                low++;
            } else {
                high--;
            }
        }
        
        
        //Brute force
        
        // int max = Integer.MIN_VALUE;
        // for(int i=0;i<height.length;i++) {
        //     for(int j=i+1;j<height.length;j++) {
        //         max = Math.max(max, (j-i)*Math.min(height[i], height[j]));
        //     }
        // }
        
        return max;
    }
}